openedx_tagging/core/tagging/models/base.py

```diff
@@ -385,6 +385,11 @@ def tag_object(
         Raised ValueError if the proposed tags are invalid for this taxonomy.
         Preserves existing (valid) tags, adds new (valid) tags, and removes omitted (or invalid) tags.
         """
+        if not isinstance(tags, (list, tuple)):
+            raise ValueError(
+                _(f"Tags must be a list or tuple, not {type(tags).__name__}.")
+            )
+        tags = list(dict.fromkeys(tags)) # Remove duplicates preserving order
```

Comment:
Removing duplicates to clean lists like `["Tag 1", "Tag 1"]` instead of throwing an `IntegrityError` on save

Reply:
:+1: